NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# DD Add Category 10

Pull Request Template


# Description

In the Category list, a user will now see a button to create a new category. When they click this, they are routed to the Category form (only one field, which is Category Name). They enter a name and hit save. The Category should be added to their database and the user should route back to the Category List, where they can see their new Category.

**Created the following files and methods:**

- In CategoryRepository.cs, ICategoryRepository.cs: added Add() method
- In CategoryController.cs: added Post() method 
- In CategoryManager.js: added addCategory() function
- CategoryForm.js

**Related Ticket(s)**
https://trello.com/c/RLkgSnrz/10-10-create-a-category

Ticket 10 Create a Category

**Type of Change**


# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `DD-addCategory-10` as one of the available branches
- `git checkout DD-addCategory-10` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin DD-addCategory-10`
- `Start Tabloid.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)
- Start Tabloid.slc by clicking the green "play" button in Visual Studio. This will open up a Swagger page with API calls.
- Run the program by navigating in a terminal TabloidFullStack > Tabloid Full Stack > client > tabloid and entering the command "npm start"


- Login to Tabloid using email "foo@bar.com" (no password needed)
- You should be presented with home screen reading [hello]. Click "Category Management" in the topside Navbar
- You should be presented with a table displaying all Categories from your database in alphabetical order
- Click "Create Category" button. You should be routed to "localhost:3000/categories/form" and see the "Create a new Category" header, as well as an input box labeled "Category Name:" and a "Save" button
- Enter a new Category Name in the input box (example: Puppies)
- Click "Save". You should be rerouted back to the Category List
- Testing is successful if the new Category Name you entered now displays with other categories in its appropriate alphabetical space. A new Id should be automatically generated that is incremented one higher from the previous highest Id

# Checklist:
